src/rashdf/geom.py

```diff
@@ -43,13 +48,97 @@ def mesh_cell_polygons(self) -> GeoDataFrame:
         GeoDataFrame
             A GeoDataFrame containing the 2D flow mesh cell polygons.
         """
-        raise NotImplementedError
+        # recursive coord cleaning
+        def clean_coords(end_coord: list = None) -> None:
+            if not end_coord:
+                target = raw_coords.pop(0)
+            else:
+                i = -1
+                for face in raw_coords:
+                    i+=1
+                    if end_coord == face[0]:
+                        target = raw_coords.pop(i)
+                    elif end_coord == face[-1]:
+                        target = raw_coords.pop(i)
+                        target=list(reversed(target))
+            [cleaned_coords.append(coord) for coord in target]
+            if raw_coords: clean_coords(target[-1])
+        def polygonize(face_coord_list) -> Polygon:
+            global cleaned_coords; global raw_coords
```

Comment:
I'm not necessarily opposed to functions-within-functions, but we should eliminate any globals. May need to rethink how this part is implemented.

Reply:
Does this achieve the same result without `global`?

```python
def clean_coords(raw_coords: list, end_coord: Optional[list] = None) -> list:
    cleaned_coords = []
    if not end_coord:
        target = raw_coords.pop(0)
    else:
        i = -1
        for face in raw_coords:
            i+=1
            if end_coord == face[0]:
                target = raw_coords.pop(i)
            elif end_coord == face[-1]:
                target = raw_coords.pop(i)
                target=list(reversed(target))
    for coord in target:
        cleaned_coords.append(coord)
    if raw_coords:
        return clean_coords(cleaned_coords, target[-1])
    return cleaned_coords

def polygonize(face_coord_list) -> Polygon:
    cleaned_coords = clean_coords(face_coord_list)
    return Polygon(cleaned_coords)
```

Reply:
Updated this method to be much simpler and easy to read -> f65b01aa50a060a798503b144f4922cee63e5d8c